src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -268,6 +270,7 @@ private static RequestDelegateFactoryContext CreateFactoryContext(RequestDelegat
 
         var serviceProvider = options?.ServiceProvider ?? options?.EndpointBuilder?.ApplicationServices ?? EmptyServiceProvider.Instance;
         var endpointBuilder = options?.EndpointBuilder ?? new RDFEndpointBuilder(serviceProvider);
+        var jsonSerializerOptions = serviceProvider.GetService<IOptions<JsonOptions>>()?.Value.SerializerOptions;
```

Comment:
Should this line fallback to `JsonOptions.DefaultSerializerOptions`? If there are no options then this will resolve to null and the method will still attempt to fetch options per request.

Or did you exclude that because that property is internal?

Reply:
Seems like an edge-case, I believe any `IOptions` type will automatically be created so this will never be null in the built-in DI case: https://source.dot.net/#Microsoft.Extensions.Options/OptionsServiceCollectionExtensions.cs,26

Reply:
Ok. And I see that `JsonOptions.SerializerOptions` has a default value so it won't be null.